plugin/evm/mempool.go

```diff
@@ -139,19 +144,40 @@ func (m *Mempool) atomicTxGasPrice(tx *Tx) (uint64, error) {
 }
 
 func (m *Mempool) Add(tx *GossipAtomicTx) error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
```

Comment:
This grabs the locks in the wrong order. The context lock must be grabbed first

Reply:
I'm still trying to figure out why we acquire the context lock here

Reply:
for future readers this is to lock state while `vm.verifyAtTip` is being run.